### PR TITLE
Remove the need for private_dns provider

### PR DESCRIPTION
**Pull request by thomast1906**
### Change description ###
Remove the need for private_dns provider

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: private-endpoint.tf [hunk 1]
```diff
@@ -10,17 +10,10 @@ resource "azurerm_private_endpoint" "postgres" {
...
-resource "azurerm_private_dns_a_record" "postgres" {
-  provider = azurerm.private_dns
-
-  name                = azurerm_postgresql_server.postgres-paas.name
-  zone_name           = "privatelink.postgres.database.azure.com"
-  resource_group_name = "core-infra-intsvc-rg"
-  ttl                 = 300
-  records             = [azurerm_private_endpoint.postgres[0].private_service_connection[0].private_ip_address]
 
+ private_dns_zone_group {
```
Comment (timja) [inline private-endpoint.tf:14]: can you also remove https://github.com/hmcts/cnp-module-postgres/blob/postgresql_tf/provider.tf#L1-L3

Comment (thomast1906) [reply private-endpoint.tf:14]: ^ changes updated

File: example/main.tf [hunk 2]
```diff
@@ -1,13 +1,12 @@
 data "azurerm_subnet" "postgres" {
-  name                 = "aks"
+  name                 = "core-infra-subnet-0-${var.env}"
   resource_group_name  = "core-infra-${var.env}"
   virtual_network_name = "core-infra-vnet-${var.env}"
 }
 
 module "db" {
   providers = {
```
Comment (timja) [inline example/main.tf:8]: this whole providers block should be removed

Comment (thomast1906) [reply example/main.tf]: Done - should have removed it

**Review by timja — APPROVED** [review]

**Pull request merged**